controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -124,3 +130,283 @@ func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocs
+											Key:                  "ceph-username",
+										},
+									},
+								},
+								{
+									Name: "ROOK_CEPH_SECRET",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{Name: "rook-ceph-mon"},
+											Key:                  "ceph-secret",
+										},
+									},
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "ceph-config", MountPath: "/etc/ceph"},
+								{Name: "mon-endpoint-volume", MountPath: "/etc/rook"},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: &[]bool{false}[0],
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{
+										"ALL",
+									},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "script",
+							Image: rookImage,
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "ceph-config", MountPath: "/etc/ceph", ReadOnly: true},
+							},
+							Command: []string{
+								"bash",
+								"-c",
+								`echo "starting configuration of mirror logging"
+								ceph config set client.rbd-mirror.a debug_ms 1
+								ceph config set client.rbd-mirror.a debug_rbd 15
+								ceph config set client.rbd-mirror.a debug_rbd_mirror 30
+								ceph config set client.rbd-mirror.a log_file /var/log/ceph/\$cluster-\$name.log
+								ceph config set client.rbd-mirror-peer debug_ms 1
+								ceph config set client.rbd-mirror-peer debug_rbd 15
+								ceph config set client.rbd-mirror-peer debug_rbd_mirror 30
+								ceph config set client.rbd-mirror-peer log_file /var/log/ceph/\$cluster-\$name.log
+								ceph config set mgr mgr/rbd_support/log_level debug
+								echo "completed"
+								`,
+							},
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: &[]bool{false}[0],
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{
+										"ALL",
+									},
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{Name: "ceph-config", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
+						{Name: "mon-endpoint-volume", VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{Name: "rook-ceph-mon-endpoints"},
+								Items:                []corev1.KeyToPath{{Key: "data", Path: "mon-endpoints"}},
+							},
+						},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: pointer.BoolPtr(true),
+						SeccompProfile: &corev1.SeccompProfile{
+							Type: corev1.SeccompProfileTypeRuntimeDefault,
+						},
+					},
+				},
+			},
+		},
+	}
+	return debugJob
+}
+
+func undoRbdMirrorDebugJob(sc *ocsv1.StorageCluster) *batchv1.Job {
```

Comment:
Avoid code duplication by defining a common job for both cases and having different functions to define only ceph arguments.